desc/compute/_field.py

```diff
@@ -2203,6 +2263,32 @@ def _B_sub_zeta_rz(params, transforms, profiles, data, **kwargs):
     return data
 
 
+@register_compute_fun(
+    name="<|B|>_axis",
+    label="\\lange |\\mathbf{B}| \\rangle_{axis}",
+    units="T",
+    units_long="Tesla",
+    description="Average magnitude of magnetic field on the magnetic axis",
+    dim=0,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    coordinates="",
+    data=["|B|", "sqrt(g)"],
+    axis_limit_data=["sqrt(g)_r"],
+)
+def _B_mag_axis(params, transforms, profiles, data, **kwargs):
```

Comment:
does this need a `resolution_requirement`?

Reply:
It's a scalar quantity (`dim=0`) so it will always be computed with a `QuadratureGrid`, but I'll add `"z"` as a resolution requirement

Reply:
We only automatically compute quantities with associated best grids when `resolution_requirement` is set and `coordinates` is such that it can be interpolated back in a lossless manner.

You'll need poloidal resolution on the innermost flux surface too. (FYI the derivative of the 3d volume jacobian $\sqrt(g)_r$ will be single valued in theta on axis, but the derivatives of the 2d ones will not).